### PR TITLE
[lldb] Add a log level to Host::SystemLog

### PR DESCRIPTION
**Pull request by JDevlieghere**
Add the ability to specify a log level to Host::SystemLog.

### CODE REVIEW

File: lldb/source/Host/common/Host.cpp [hunk 1]
```diff
@@ -91,15 +91,37 @@ using namespace lldb_private;
...
+  int level = LOG_DEBUG;
+  switch (log_level) {
+  case eSystemLogInfo:
+    level = LOG_INFO;
+    break;
+  case eSystemLogWarning:
+    level = LOG_WARNING;
+    break;
+  case eSystemLogError:
+    level = LOG_WARNING;
```
Comment (medismailben) [inline lldb/source/Host/common/Host.cpp:108]: typo ?

Comment (medismailben) [inline lldb/source/Host/common/Host.cpp:108]: typo ?

```suggestion
    level = LOG_ERR;
```

**Review by medismailben — APPROVED** [review]
LGTM!

File: lldb/include/lldb/Host/Host.h [hunk 2]
```diff
@@ -87,8 +87,15 @@ class Host {
   StartMonitoringChildProcess(const MonitorChildProcessCallback &callback,
                               lldb::pid_t pid);
 
+  /// System log level.
+  enum SystemLogLevel {
+    eSystemLogInfo,
+    eSystemLogWarning,
```
Comment (adrian-prantl) [inline lldb/include/lldb/Host/Host.h:93]: Is there a good reason to keep this separate from DiagnosticSeverity?
https://lldb.llvm.org/cpp_reference/namespacelldb__private.html#a3b3fff978827f40d27f288b4f1f29369
Maybe we could factor this into private enums?

Comment (JDevlieghere) [reply lldb/include/lldb/Host/Host.h:93]: Good point! https://github.com/llvm/llvm-project/pull/90917

**Pull request merged**